### ISSUE TITLE
Unable to find a soft-roce device

### ISSUE THREAD

**Issue by Yahoo-90** [openucx/ucx#9207]
### Describe the bug
   I have completed the installation of mpi, ucx, and soft-roce in the Docker container, and tested the connectivity of the soft-roce devices through the 'perftest'. However, when I use ucx, it shows that the soft-roce devices cannot be found. May I ask if ucx supports soft-roce, and if so, what should I do to enable ucx to find that virtual device.

### Steps to Reproduce
- Command line
- UCX version used (from github branch XX or release YY) + UCX configure flags (can be checked by `ucx_info -v`)
   Library version: 1.16.0
   Library path: /home/sunshuo/ucx/lib/libucs.so.0
   API headers version: 1.16.0
   Git branch 'master', revision 4725537

- **Any UCX environment variables used**

### Setup and versions
- OS version (Linux distro) + CPU architecture (x86_64)
- For RDMA/IB/RoCE related issues:
    - Driver version:
        - rdma-core libibverbs
   - HW information from `ibstat` or `ibv_devinfo -vv` command
      hca_id:	rxe0
	transport:			InfiniBand (0)
	fw_ver:				0.0.0
	node_guid:			ee2a:72ff:fe22:f4a2
	sys_image_guid:			ee2a:72ff:fe22:f4a2
	vendor_id:			0x0000
	vendor_part_id:			0
	hw_ver:				0x0
	phys_port_cnt:			1
	max_mr_size:			0xffffffffffffffff
	page_size_cap:			0xfffff000
	max_qp:				65536
	max_qp_wr:			16384
	device_cap_flags:		0x00203c76
					BAD_PKEY_CNTR
					BAD_QKEY_CNTR
					AUTO_PATH_MIG
					CHANGE_PHY_PORT
					UD_AV_PORT_ENFORCE
					PORT_ACTIVE_EVENT
					SYS_IMAGE_GUID
					RC_RNR_NAK_GEN
					SRQ_RESIZE
					MEM_MGT_EXTENSIONS
	max_sge:			32
	max_sge_rd:			32
	max_cq:				16384
	max_cqe:			32767
	max_mr:				262144
	max_pd:				32764
	max_qp_rd_atom:			128
	max_ee_rd_atom:			0
	max_res_rd_atom:		258048
	max_qp_init_rd_atom:		128
	max_ee_init_rd_atom:		0
	atomic_cap:			ATOMIC_HCA (1)
	max_ee:				0
	max_rdd:			0
	max_mw:				0
	max_raw_ipv6_qp:		0
	max_raw_ethy_qp:		0
	max_mcast_grp:			8192
	max_mcast_qp_attach:		56
	max_total_mcast_qp_attach:	458752
	max_ah:				100
	max_fmr:			0
	max_srq:			960
	max_srq_wr:			16384
	max_srq_sge:			27
	max_pkeys:			64
	local_ca_ack_delay:		15
	general_odp_caps:
	rc_odp_caps:
					NO SUPPORT
	uc_odp_caps:
					NO SUPPORT
	ud_odp_caps:
					NO SUPPORT
	xrc_odp_caps:
					NO SUPPORT
	completion_timestamp_mask not supported
	core clock not supported
	device_cap_flags_ex:		0x0
	tso_caps:
	max_tso:			0
	rss_caps:
		max_rwq_indirection_tables:			0
		max_rwq_indirection_table_size:			0
		rx_hash_function:				0x0
		rx_hash_fields_mask:				0x0
	max_wq_type_rq:			0
	packet_pacing_caps:
		qp_rate_limit_min:	0kbps
		qp_rate_limit_max:	0kbps
	tag matching not supported
		port:	1
			state:			PORT_ACTIVE (4)
			max_mtu:		4096 (5)
			active_mtu:		1024 (3)
			sm_lid:			0
			port_lid:		0
			port_lmc:		0x00
			link_layer:		Ethernet
			max_msg_sz:		0x800000
			port_cap_flags:		0x00010000
			port_cap_flags2:	0x0000
			max_vl_num:		1 (1)
			bad_pkey_cntr:		0x0
			qkey_viol_cntr:		0x0
			sm_sl:			0
			pkey_tbl_len:		64
			gid_tbl_len:		1024
			subnet_timeout:		0
			init_type_reply:	0
			active_width:		1X (1)
			active_speed:		2.5 Gbps (1)
			phys_state:		LINK_UP (5)
			GID[  0]:		fe80:0000:0000:0000:ee2a:72ff:fe22:f4a2
			GID[  1]:		0000:0000:0000:0000:0000:ffff:c0a8:c7cc
			GID[  2]:		fe80:0000:0000:0000:8286:f59f:3ecf:c81f

### Additional information (depending on the issue)
- OpenMPI version:4.1.4
- Output of `ucx_info -d` to show transports and devices recognized by UCX:
  Memory domain: self
     Component: self
             register: unlimited, cost: 0 nsec
           remote key: 0 bytes
           rkey_ptr is supported
         memory types: host (access,reg_nonblock,reg,cache)

      Transport: self
         Device: memory
           Type: loopback
  System device: <unknown>

      capabilities:
            bandwidth: 0.00/ppn + 19360.00 MB/sec
              latency: 0 nsec
             overhead: 10 nsec
            put_short: <= 4294967295
            put_bcopy: unlimited
            get_bcopy: unlimited
             am_short: <= 8K
             am_bcopy: <= 8K
               domain: cpu
           atomic_add: 32, 64 bit
           atomic_and: 32, 64 bit
            atomic_or: 32, 64 bit
           atomic_xor: 32, 64 bit
          atomic_fadd: 32, 64 bit
          atomic_fand: 32, 64 bit
           atomic_for: 32, 64 bit
          atomic_fxor: 32, 64 bit
          atomic_swap: 32, 64 bit
         atomic_cswap: 32, 64 bit
           connection: to iface
      device priority: 0
     device num paths: 1
              max eps: inf
       device address: 0 bytes
        iface address: 8 bytes
       error handling: ep_check

 Memory domain: tcp
     Component: tcp
             register: unlimited, cost: 0 nsec
           remote key: 0 bytes
         memory types: host (access,reg_nonblock,reg,cache)

      Transport: tcp
         Device: eno8303
           Type: network
  System device: <unknown>

      capabilities:
            bandwidth: 113.16/ppn + 0.00 MB/sec
              latency: 5776 nsec
             overhead: 50000 nsec
            put_zcopy: <= 18446744073709551590, up to 6 iov
  put_opt_zcopy_align: <= 1
        put_align_mtu: <= 0
             am_short: <= 8K
             am_bcopy: <= 8K
             am_zcopy: <= 64K, up to 6 iov
   am_opt_zcopy_align: <= 1
         am_align_mtu: <= 0
            am header: <= 8037
           connection: to ep, to iface
      device priority: 0
     device num paths: 1
              max eps: 256
       device address: 6 bytes
        iface address: 2 bytes
           ep address: 10 bytes
       error handling: peer failure, ep_check, keepalive

      Transport: tcp
         Device: docker0
           Type: network
  System device: <unknown>

      capabilities:
            bandwidth: 11.32/ppn + 0.00 MB/sec
              latency: 10960 nsec
             overhead: 50000 nsec
            put_zcopy: <= 18446744073709551590, up to 6 iov
  put_opt_zcopy_align: <= 1
        put_align_mtu: <= 0
             am_short: <= 8K
             am_bcopy: <= 8K
             am_zcopy: <= 64K, up to 6 iov
   am_opt_zcopy_align: <= 1
         am_align_mtu: <= 0
            am header: <= 8037
           connection: to ep, to iface
      device priority: 1
     device num paths: 1
              max eps: 256
       device address: 6 bytes
        iface address: 2 bytes
           ep address: 10 bytes
       error handling: peer failure, ep_check, keepalive

      Transport: tcp
         Device: lo
           Type: network
  System device: <unknown>

      capabilities:
            bandwidth: 11.91/ppn + 0.00 MB/sec
              latency: 10960 nsec
             overhead: 50000 nsec
            put_zcopy: <= 18446744073709551590, up to 6 iov
  put_opt_zcopy_align: <= 1
        put_align_mtu: <= 0
             am_short: <= 8K
             am_bcopy: <= 8K
             am_zcopy: <= 64K, up to 6 iov
   am_opt_zcopy_align: <= 1
         am_align_mtu: <= 0
            am header: <= 8037
           connection: to ep, to iface
      device priority: 1
     device num paths: 1
              max eps: 256
       device address: 18 bytes
        iface address: 2 bytes
           ep address: 10 bytes
       error handling: peer failure, ep_check, keepalive

 Connection manager: tcp
      max_conn_priv: 2064 bytes

 Memory domain: sysv
     Component: sysv
             allocate: unlimited
           remote key: 12 bytes
           rkey_ptr is supported
         memory types: host (access,alloc,cache)

      Transport: sysv
         Device: memory
           Type: intra-node
  System device: <unknown>

      capabilities:
            bandwidth: 0.00/ppn + 15360.00 MB/sec
              latency: 80 nsec
             overhead: 10 nsec
            put_short: <= 4294967295
            put_bcopy: unlimited
            get_bcopy: unlimited
             am_short: <= 100
             am_bcopy: <= 8256
               domain: cpu
           atomic_add: 32, 64 bit
           atomic_and: 32, 64 bit
            atomic_or: 32, 64 bit
           atomic_xor: 32, 64 bit
          atomic_fadd: 32, 64 bit
          atomic_fand: 32, 64 bit
           atomic_for: 32, 64 bit
          atomic_fxor: 32, 64 bit
          atomic_swap: 32, 64 bit
         atomic_cswap: 32, 64 bit
           connection: to iface
      device priority: 0
     device num paths: 1
              max eps: inf
       device address: 16 bytes
        iface address: 8 bytes
       error handling: ep_check

 Memory domain: posix
     Component: posix
             allocate: <= 64M
           remote key: 32 bytes
           rkey_ptr is supported
         memory types: host (access,alloc,cache)

      Transport: posix
         Device: memory
           Type: intra-node
  System device: <unknown>

      capabilities:
            bandwidth: 0.00/ppn + 15360.00 MB/sec
              latency: 80 nsec
             overhead: 10 nsec
            put_short: <= 4294967295
            put_bcopy: unlimited
            get_bcopy: unlimited
             am_short: <= 100
             am_bcopy: <= 8256
               domain: cpu
           atomic_add: 32, 64 bit
           atomic_and: 32, 64 bit
            atomic_or: 32, 64 bit
           atomic_xor: 32, 64 bit
          atomic_fadd: 32, 64 bit
          atomic_fand: 32, 64 bit
           atomic_for: 32, 64 bit
          atomic_fxor: 32, 64 bit
          atomic_swap: 32, 64 bit
         atomic_cswap: 32, 64 bit
           connection: to iface
      device priority: 0
     device num paths: 1
              max eps: inf
       device address: 16 bytes
        iface address: 16 bytes
       error handling: ep_check

 Memory domain: cma
     Component: cma
             register: unlimited, cost: 9 nsec
         memory types: host (access,reg_nonblock,reg,cache)

      Transport: cma
         Device: memory
           Type: intra-node
  System device: <unknown>

      capabilities:
            bandwidth: 0.00/ppn + 11145.00 MB/sec
              latency: 80 nsec
             overhead: 2000 nsec
            put_zcopy: unlimited, up to 16 iov
  put_opt_zcopy_align: <= 1
        put_align_mtu: <= 1
            get_zcopy: unlimited, up to 16 iov
  get_opt_zcopy_align: <= 1
        get_align_mtu: <= 1
           connection: to iface
      device priority: 0
     device num paths: 1
              max eps: inf
       device address: 16 bytes
        iface address: 16 bytes
       error handling: peer failure, ep_check

**Comment by yosefe** [comment]
@Yahoo-90 UCX was never tested on soft-roce.
Can you pls [build UCX with debug info](https://github.com/openucx/ucx/wiki/Logging) and run `env UCX_LOG_LEVEL=data ucx_info -d`, to see where the device gets disabled?
Thanks!

**Comment by Yahoo-90** [comment]
@yosefe I bound my soft-roce device to eno8303 and entered this command（env UCX_LOG_LEVEL=data ucx_info -d）. The result is shown in the file below. I did not see anything related to rdma/roce in the result. But when I use ibv_ devices，the virtual device is displayed like the picture below. Does this mean that ucx cannot recognize this virtual device?
[debug_info.txt](https://github.com/openucx/ucx/files/12010853/debug_info.txt)
![ibv_devices](https://github.com/openucx/ucx/assets/65900256/c7c93130-815e-45d9-b2cf-9d43a511f9aa)

**Comment by yosefe** [comment]
@Yahoo-90 it seems UCX was built without RDMA/Verbs support.

**Comment by Yahoo-90** [comment]
@yosefe I remember when I ran  ./configure --prefix=/xxx, the final run result shows that there are no components in the IB modules section. What should I do to enable ucx to download RDMA support?

**Comment by Yahoo-90** [comment]
@I'm sorry for the late response. I was worried about the server environment, so I performed the same operation on the virtual machine, and the result was the same. ucx did not install the IB module in the configure step. The following file is the result of configuring. I downloaded rdma core directly using apt-get, and I believe its version should be the latest or near the latest, at least much greater than version 28.0.
[configure_result.txt](https://github.com/openucx/ucx/files/12027222/configure_result.txt)
